File: javaSrc/src/ice/type/IceItem.java
package ice.type;

import ice.ui.stat.IceStat;
import ice.ui.stat.IceStats;
import arc.graphics.Color;
import mindustry.type.Item;

public class IceItem extends Item {
    public float nutrientConcentration = 0;

    public IceItem(String name) {
        this(name, new Color(Color.black));
    }

    public IceItem(String name, Color color) {
        super(name);
        this.color = color;
    }

    public IceItem(String name, String color) {
        super(name);
        this.color = Color.valueOf(color);
    }

    public IceItem(String name, String color, ItemLambda lambda) {
        super(name);
        lambda.get(this);
        this.color = Color.valueOf(color);
    }

    @Override
    public void init() {
        stats = new IceStats();
        super.init();
    }


    @Override
    public void setStats() {
        IceStats stats1 = (IceStats) stats;
        if (nutrientConcentration != 0) {
            stats1.addPercentThrob(IceStat.nutrientConcentration, nutrientConcentration, "[red]");
        }
        super.setStats();
    }

    public interface ItemLambda {
        IceItem get(IceItem item);
    }
}

File: src/ice/library/baseContent/item/IceItem.kt
package ice.library.baseContent.item

import arc.graphics.Color
import ice.content.IPlanets
import ice.library.components.NutrientConcentration
import ice.library.meta.stat.IceStats
import mindustry.type.Item

open class IceItem(name: String, color: String) : Item(name, Color.valueOf(color)), NutrientConcentration {
    var nutrientConcentration = 0f
    override fun postInit() {
        shownPlanets.add(IPlanets.阿德里)
        super.postInit()
    }

    override fun setStats() {
        super.setStats()
        if (nutrientConcentration != 0f) stats.addPercent(IceStats.营养浓度, nutrientConcentration)
        stats.add(IceStats.建造时间花费, cost)
        stats.add(IceStats.建筑血量系数, healthScaling)
        stats.add(IceStats.硬度, "$hardness")
        stats.add(IceStats.是否用于建造, buildable)
    }

    override fun getNutrient(): Float = nutrientConcentration
}
